homeassistant/components/freebox/router.py

```diff
@@ -0,0 +1,247 @@
+"""Represent the Freebox router and its devices and sensors."""
+from datetime import datetime, timedelta
+import logging
+import os
+from typing import Dict
+
+from aiofreepybox import Freepybox
+from aiofreepybox.api.wifi import Wifi
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.util import slugify
+
+from .const import (
+    API_VERSION,
+    APP_DESC,
+    CONNECTION_SENSORS,
+    DEFAULT_DEVICE_NAME,
+    DOMAIN,
+    SENSOR_NAME,
+    SENSOR_UPDATE,
+    STORAGE_KEY,
+    STORAGE_VERSION,
+    TEMPERATURE_SENSOR_TEMPLATE,
+    TRACKER_UPDATE,
+)
+from .device_tracker import FreeboxDevice
+from .sensor import FreeboxSensor
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+
+class FreeboxRouter:
+    """Representation of a Freebox router."""
+
+    def __init__(self, hass: HomeAssistantType, entry: ConfigEntry):
+        """Initialize a Freebox router."""
+        self.hass = hass
+        self._entry = entry
+        self._host = entry.data[CONF_HOST]
+        self._port = entry.data[CONF_PORT]
+
+        self._api: Freepybox = None
+        self._name = None
+        self._mac = None
+        self._sw_v = None
+
+        # Devices
+        self._devices: Dict[str, FreeboxDevice] = {}
+
+        # Sensors
+        self._temperature_sensors: Dict[str, FreeboxSensor] = {}
```

Comment:
Don't store the entity instances. It's ok to store an id reference to the entity.